File: lcof/of005/Solution.go
package Solution

func replaceSpace(s string) string {
	ans := make([]rune, len(s)*3)

	i := 0
	for _, v := range s {
		if v != ' ' {
			ans[i] = v
			i++
		} else {
			ans[i], ans[i+1], ans[i+2] = '%', '2', '0'
			i += 3
		}
	}
	return string(ans)[:i]
}

File: lcof/of005/Solution.py
class Solution:
    def replaceSpace(self, s: str) -> str:
        ans = []
        for v in s:
            if v == ' ':
                ans.append('%20')
            else:
                ans.append(v)
        return ''.join(ans)


class Solution2:
    def replaceSpace(self, s: str) -> str:
        return s.replace(' ', '%20')

if __name__ == '__main__':
    print(Solution().replaceSpace("We are happy."))
    print(Solution2().replaceSpace("We are happy."))
